core/capabilities/ccip/common/addresscodec.go

```diff
@@ -4,41 +4,20 @@ import (
 	"fmt"
 
 	chainsel "github.com/smartcontractkit/chain-selectors"
-
 	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 )
 
-// ChainSpecificAddressCodec is an interface that defines the methods for encoding and decoding addresses
-type ChainSpecificAddressCodec interface {
-	AddressBytesToString([]byte) (string, error)
-	AddressStringToBytes(string) ([]byte, error)
-}
-
 // AddressCodec is a struct that holds the chain specific address codecs
 type AddressCodec struct {
 	EVMAddressCodec    ChainSpecificAddressCodec
 	SolanaAddressCodec ChainSpecificAddressCodec
 }
 
-// AddressCodecParams is a struct that holds the parameters for creating a AddressCodec
-type AddressCodecParams struct {
-	evmAddressCodec    ChainSpecificAddressCodec
-	solanaAddressCodec ChainSpecificAddressCodec
-}
-
-// NewAddressCodecParams is a constructor for AddressCodecParams
-func NewAddressCodecParams(evmAddressCodec ChainSpecificAddressCodec, solanaAddressCodec ChainSpecificAddressCodec) AddressCodecParams {
-	return AddressCodecParams{
-		evmAddressCodec:    evmAddressCodec,
-		solanaAddressCodec: solanaAddressCodec,
-	}
-}
-
-// NewAddressCodec is a constructor for AddressCodec
-func NewAddressCodec(params AddressCodecParams) AddressCodec {
+// NewAddressCodec is a constructor for NewAddressCodec
+func NewAddressCodec(evmAddrCodec, solanaAddrCodec ChainSpecificAddressCodec) AddressCodec {
```

Comment:
Could we try to not force the caller of these to also provide implementations for each chain.
I mean not having to pass params for evmAddrCodec, solanaAddrCodec, but instead just instantiate them inside this New... function.

Same for all other codecs or chain specific types.

The reason is because we want all the different chain implementations to be instantiated from 1 file/package. The common package seems fine for that.

You might face circular dep, which can be resolved by moving all the interfaces to a separate package.
capabilities/ccip/types. This way, the chain implementations can reference the interface they implement, and provide a default instance too.

Reply:
The circular dependency comes from the constructor such as `ccipcommon.NewExtraDataCodec` is called by `executecodec_test.go` from `ccip/ccipevm` for example. And we can't have `ccip/common` import `ccip/ccipevm` again.

Reply:
So even if I move the interface to ccip/types, this will still be an issue if I reference ccipevm or ccipsolana during initialization in the ccip/common. But I agree it would be more ideal to have a more elegant way to avoid modifying the caller when we add a new chain.

Reply:
I applied the default config approach from @winder's [PR](https://github.com/smartcontractkit/chainlink/pull/17173)